syntax = "proto3";
package tzrpc.proto;
import "String.proto";
import "Number.proto";
import "Bytes.proto";
import "Boolean.proto";
import "Numpy.proto";
import "Tensor.proto";

service toObject {
  rpc toString(String) returns (String) {};
  rpc toInteger(String) returns (Integer) {};
  rpc toFloat(String) returns (Float) {};
  rpc toDouble(String) returns (Double) {};
  rpc toBoolean(String) returns (Boolean) {};
  rpc toBytes(String) returns (Bytes) {};
  rpc toNdarray(String) returns (ndarray) {};
  rpc toTensor(String) returns (Tensor) {};

  rpc toNdarrays(String) returns (ndarrays) {};
  rpc toTensors(String) returns (Tensors) {};
  rpc toIntegerArrayList(String) returns (IntegerArrayList) {};
  rpc toFloatArrayList(String) returns (FloatArrayList) {};
  rpc toDoubleArrayList(String) returns (DoubleArrayList) {};
  rpc toBooleanArrayList(String) returns (BooleanArrayList) {};
}

